dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/programrule/executor/event/SetMandatoryFieldExecutorTest.java

```diff
@@ -184,13 +185,14 @@ void shouldFailValidationWhenMandatoryFieldIsDeletedAndStrategyIsUpdate() {
 
   @ParameterizedTest
   @MethodSource("transactionsNotCreatingDataValues")
-  void shouldPassValidationWhenMandatoryFieldIsNotPresentAndStrategyIsUpdate(
+  void shouldPassValidationWhenMandatoryFieldIsNotPresentInPayloadButPresentInDB(
       EventStatus savedStatus, EventStatus newStatus) {
     when(preheat.getIdSchemes()).thenReturn(TrackerIdSchemeParams.builder().build());
     when(preheat.getDataElement(DATA_ELEMENT_UID.getValue())).thenReturn(dataElement);
     when(preheat.getProgramStage(MetadataIdentifier.ofUid(programStage))).thenReturn(programStage);
     Event event = getEventWithMandatoryValueNOTSet(savedStatus);
-    when(preheat.getEvent(event.getUid())).thenReturn(event(event.getUid(), newStatus));
+    when(preheat.getEvent(event.getUid()))
+        .thenReturn(eventWithMandatoryValue(event.getUid(), newStatus));
```

Comment:
Should that not be `savedStatus` as the preheat has the events from the DB 🤔 
and `newStatus` should be set on the `event` and its reference in the `bundle`.

Reply:
Yes :)
I changed and everything is still working because we are transitioning between statuses that allow data values, so swapping the saved and new didn't change the result.